rhobs/alerting/data_plane/prometheus.pipeline_alerts.yaml

```diff
@@ -183,3 +183,45 @@ spec:
             team: pipelines
             # link will be available once https://issues.redhat.com/browse/SRVKP-5899 is complete
             runbook_url: https://gitlab.cee.redhat.com/konflux/docs/sop/-/blob/main/pipeline-service/slos/pac-performance.md
+        - alert: PipelinesControllerPipelineRunRecLeasesNotDistributed
+          expr: count(count by (lease_holder) (kube_lease_owner{lease=~"tekton-pipelines-controller.github.com.tektoncd.pipeline.pkg.reconciler.pipelinerun.reconciler..*", lease_holder!=""})) <= 1
```

Comment:
Metric `kube_lease_owner` is not currently exported out of the cluster. Before it can be used, it needs to be exported. The instructions are here: https://gitlab.cee.redhat.com/konflux/docs/sop/-/blob/main/o11y/monitoring/tshoot-missing-metrics.md?ref_type=heads#missing-metric

Also, do you know how much does the identity of individual data series for this metric change over time? We have quota limits on the number of time series we hold in RHOBS, so we need to see that we're not exporting metrics for which new data series are added frequently based on the service workload.
Can you tell what's causing new data series to appear for that metric?

cc @amisstea @mftb

Reply:
Thank you Yftach! Created https://github.com/redhat-appstudio/infra-deployments/pull/4583. Hope it is OK I also added to `LabelKeep` regexp.

Regarding how many labels combinations this mean, on `stone-stg-rh01` during last week I see 51. There are actually only 2 labels that change over time:

* `lease` - if we do not reconfigure Tekton Pipelines Controller to use different buckets count, there will be just 4 of these for every reconciler (so 12)
* `lease_holder` - this is a Pipelines Controller pod identification, so every time that pod restarts or so, new label value appears here (these pods are supposed to be stable, always running and restarts will be matter of upgrades or signs of cluster issues)